src/control/cmd/daos_agent/fabric.go

```diff
@@ -35,7 +35,12 @@ type FabricInterface struct {
 }
 
 func (f *FabricInterface) Providers() []string {
-	return f.hw.Providers.ToSlice()
+	provs := f.hw.Providers.ToSlice()
+	provStrs := []string{}
```

Comment:
minor: You could use `provStrs := make([]string, len(provs))` and then set the indexed element in the range to `p.Name` to avoid the `append()` overhead.

Reply:
happy to do so but curious if the append overhead is significant?

Reply:
done